plugins/foreman_ansible/2.x/index.md

```diff
@@ -19,42 +19,124 @@ You can define a list of roles to enforce, or just run arbitrary playbooks on th
+    $(".ansible_version_none").show();
+  }
+}
+</script>
+
+In order to make Ansible send us data from the hosts, we set up a callback on your host that runs Ansible.
+To provide specific nstallation instructions, please select your Ansible version:
+
+<select onChange="update_ansible_version(this);">
+  <option value="none">-- select Ansible version --</option>
+  <option value="25">2.5 or higher</option>
+  <option value="22">2.2, 2.3, 2.4</option>
+  <option value="21">2.1 or lower</option>
+</select>
+
+<div class="ansible_version ansible_version_none">
+  <i>No Ansible version selected.</i>
+</div>
+
+
+<div class="ansible_version ansible_version_25 ansible_version_22" style="display:none;">
+<p>
+Our callback is installed with Ansible itself, simply change <code>/etc/ansible/ansible.cfg</code> to contain
+</p>
+
+{% highlight bash %}
+[defaults]
+callback_whitelist = foreman
+{% endhighlight %}
+
+<p>
 and the callback will be enabled after that.
-
-
-For Ansible versions older than 2.2:
-
-* Change your `/etc/ansible/ansible.cfg` to contain the following options
-
-      callback_plugins = ~/.ansible/plugins/callback_plugins/
-      bin_ansible_callbacks = True
-
-* And run this line to download the callback file in the appropriate location
-
-      wget https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/foreman.py -O ~/.ansible/plugins/callback_plugins/foreman.py
-
-* Alternatively, you can find the source for the [Ansible callback here](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/foreman.py) and deploy it in your directory of choice.
+</p>
+
+<p>
+Of course, the callback cannot know automatically where is Foreman located and which certificates to use in its requests.
+</p>
+</div>
+
+<div class="ansible_version ansible_version_25" style="display:none;">
+<p>
+To configure it, simply add a section <code>[callback_foreman]</code> at the end of <code>/etc/ansible/ansible.cfg</code> like this:
+</p>
+
+{% highlight bash %}
+[callback_foreman]
+url = 'https://centos7-devel.lobatolan.home'
+ssl_cert = /home/dlobatog/workspace/ansible-demo/katello-default-ca.crt
+ssl_key = /home/dlobatog/workspace/ansible-demo/katello-default-ca.key
+verify_certs = False
+{% endhighlight %}
+</div>
+
+<div class="ansible_version ansible_version_21" style="display:none;">
+<p>
+We don't really test these versions, however if you need to use an old version of Ansible, you could do the following:
+</p>
+
+<ul>
+  <li>
+  Change your <code>/etc/ansible/ansible.cfg</code> to contain the following options:
+  </li>
+
+{% highlight bash %}
+callback_plugins = ~/.ansible/plugins/callback_plugins/
+bin_ansible_callbacks = True
+{% endhighlight %}
+
+  <li>
+  And run this line to download the callback file in the appropriate location:
+  </li>
+
+{% highlight bash %}
+wget https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/foreman.py -O ~/.ansible/plugins/callback_plugins/foreman.py
```

Comment:
In 2.5 they changed the API of initialization so I doubt it's going to work. Probably best to take the one from 2.2 or just drop support for it. Debian Stretch includes Ansible 2.2 and EPEL includes 2.5. Only Ubuntu Xenial includes 2.0 (Xenial backports has 2.1) so that might be annoying.